src/Umbraco.Cms.Integrations.Search.Algolia/Services/AlgoliaSearchPropertyIndexValueFactory.cs

```diff
@@ -11,12 +12,16 @@ public class AlgoliaSearchPropertyIndexValueFactory : IAlgoliaSearchPropertyInde
 
         private readonly IMediaService _mediaService;
 
-        public AlgoliaSearchPropertyIndexValueFactory(IDataTypeService dataTypeService, IMediaService mediaService)
+        private readonly IEnumerable<string> _availableLanguages;
+
+        public AlgoliaSearchPropertyIndexValueFactory(IDataTypeService dataTypeService, IMediaService mediaService, ILocalizationService localizationService)
         {
             _dataTypeService = dataTypeService;
 
             _mediaService = mediaService;
 
+            _availableLanguages = localizationService.GetAllLanguages().Select(p => p.IsoCode);
```

Comment:
Is this service registered a singleton?  If so, this wouldn't get any updates to the languages created in Umbraco until a restart.  And if so. might mean this needs to be saved as `_localizationService` and the `GetAllLanguages()` method called at the time of indexing.